src/applications/disability-benefits/all-claims/tests/pages/addDisabilitiesRevised.unit.spec.jsx

```diff
@@ -188,4 +190,66 @@ describe('showRevisedNewDisabilitiesPage', () => {
       expect(showRevisedNewDisabilitiesPage(formData)).to.be.true;
     });
   });
+
+  describe('updateFormData', () => {
+    // It's a function just to make sure we're not mutating it anywhere along the way
+    const oldData = () => ({
+      newDisabilities: [{ condition: 'Something with-hyphens and ALLCAPS' }],
+      vaTreatmentFacilities: [
+        {
+          treatedDisabilityNames: {
+            somethingwithhyphensandallcaps: true,
+          },
+        },
+      ],
+      'view:isPow': {
+        powDisabilities: { somethingwithhyphensandallcaps: true },
+      },
+    });
+
+    describe('should return unmodified newData', () => {
+      it("if oldData.newDisabilities doesn't exist", () => {
+        const newData = { newDisabilities: ['foo'] };
+        expect(updateFormData({}, newData)).to.deep.equal(newData);
+      });
+      it("if newData.newDisabilities doesn't exist", () => {
+        const newData = {};
+        expect(updateFormData(oldData(), newData)).to.deep.equal(newData);
+      });
+      it('if no disabilities changed', () => {
+        const old = oldData();
+        expect(updateFormData(old, old)).to.deep.equal(old);
+      });
+    });
+
+    it('should not modify oldData', () => {
+      const old = oldData();
+      updateFormData(
+        old,
+        set('newDisabilities[1]', { condition: 'Something else' }, oldData()),
+      );
+      expect(old).to.deep.equal(oldData());
```

Comment:
Can you help me understand what this is testing?

If we have old defined as 
```
old = {
  newDisabilities: [{ condition: 'Something with-hyphens and ALLCAPS' }],
  ...
};
```
Then call set() to update to
```
newObject = {
  newDisabilities: [
    { condition: 'Something with-hyphens and ALLCAPS' },
    { condition: 'Something else' },
  ],
  ...
}
```
why would `updateFormData(old, newObject)` return old?

Reply:
it's testing that that old data object is not mutated on an update

> why would updateFormData(old, newObject) return old?
it doesn't return old, but we are testing that there are no sideeffects to existing objects.

to be clear, these tests + functions are very similar to the ones on the existing addDisabilities page, but the tests hadn't been moved over.

Reply:
I see what you're saying. Thanks for explaining.